mongo/mongo.go

```diff
@@ -212,12 +212,44 @@ type EnsureServerParams struct {
 	JujuDBSnapChannel string
 }
 
-// EnsureServer ensures that the MongoDB server is installed,
+// EnsureServerStarted ensures that the MongoDB server is installed,
 // configured, and ready to run.
-//
-// This method will remove old versions of the mongo init service as necessary
-// before installing the new version.
-func EnsureServer(args EnsureServerParams) error {
+func EnsureServerStarted(snapChannel string) error {
+	return ensureServerStarted(dataPathForJujuDbSnap, snapChannel)
+}
+
+func ensureServerStarted(dataDir, snapChannel string) error {
+	svc, err := mongoSnapService(dataDir, systemd.EtcSystemdDir, snapChannel)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	installed, err := svc.Installed()
+	if err != nil {
+		// If not installed, returns error "exit status 1"
+		// When run on the cli and not installed:
+		//    error: snap "juju-db" not found
+		// However juju doesn't get a typed error in return.
+		// Log it and continue.
+		logger.Debugf("installed returned %s", err)
+	}
+	if !installed {
+		return errors.NotFoundf("mongo service %q", svc.Name())
+	}
+	running, err := svc.Running()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if !running {
+		if err = svc.Start(); err != nil {
```

Comment:
Is starting synchronous, that is, I guess we don't need to check `svc.Running()` again after starting it?

Reply:
internally Start() calls `running, err := s.Running()`
During testing, if you change the memory value and immediately run juju status, there's a slight pause as the controller re-establishes connection and the workers restart